pkg/kdk/restart.go

```diff
@@ -0,0 +1,52 @@
+// Copyright © 2018 Cisco Systems, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kdk
+
+import (
+	"github.com/ghodss/yaml"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"strings"
+)
+
+func Restart(cfg KdkEnvConfig) {
+
+	log.Info("Restarting KDK container")
+
+	// Create snapshot of running KDK container
+	snapshotName, err := Snapshot(cfg)
+	if err != nil {
+		log.WithField("err", err).Fatal("Failed to create KDK image snapshot")
+	}
+
+	// Destroy running KDK container
+	Destroy(cfg)
+
+	// Save config with snapshot image tag
+	cfg.ConfigFile.AppConfig.ImageTag = strings.Split(snapshotName, ":")[1]
+	cfg.ConfigFile.ContainerConfig.Image = snapshotName
+	y, err := yaml.Marshal(cfg.ConfigFile)
```

Comment:
Once you do a restart it creates a snapshot and then overwrites the config file so that after a kdk destroy one will not access  a default expected image again.  Let's fix this so that after a kdk shapshot, a kdk destroy and then a subsequent kdk up/ssh will create a new kdk with the default image.

Reply:
Fixed